artman/tasks/python_grpc_tasks.py

```diff
@@ -94,35 +94,41 @@ def _extract_base_dirs(self, proto_file):
             return ''
 
     def _transform(self, pkg, sep, common_protos):
-        """Add 'proto' package after 'google' or 'google.cloud'
+        """Transform to the appropriate proto package layout.
 
         Works with arbitrary separator (e.g., '/' for import statements,
         '.' for proto package statements, os.path.sep for filenames)
         """
-        # Skip common protos
-        pkg_list = pkg.split(sep)
-
-        dotted = '.'.join(pkg_list)
+        if sep != '.' and pkg.endswith('.proto'):
+            dotted = pkg[:-6].replace(sep, '.')
+            suffix = '.proto'
+        else:
+            dotted = pkg.replace(sep, '.')
+            suffix = ''
+
+        # Sanity check: Do not transform common protos.
         for common_pkg in common_protos:
             if dotted.startswith(common_pkg):
-                return sep.join(pkg_list)
+                return pkg
+
+        # Transform into the ideal proto path.
```

Comment:
I don't know the knowledge about why this transforming is needed. But if this is going to work when people do `pip install`, I am okay with any transforming.